TM1py/Services/CellService.py

```diff
@@ -602,19 +604,16 @@ def write(self, cube_name: str, cellset_as_dict: Dict, dimensions: Iterable[str]
                 deactivate_transaction_log=deactivate_transaction_log,
                 reactivate_transaction_log=reactivate_transaction_log,
                 precision=precision,
+                skip_non_updateable=skip_non_updateable,
                 **kwargs)
 
         if not dimensions:
             dimensions = self.get_dimension_names_for_writing(cube_name=cube_name, **kwargs)
 
-        values = []
-        query = MdxBuilder.from_cube(cube_name)
-        for coordinates, value in cellset_as_dict.items():
-            members = (Member.of(dimension, element) for dimension, element in zip(dimensions, coordinates))
-            query.add_member_tuple_to_columns(*members)
-            values.append(value)
-        mdx = query.to_mdx()
+        if skip_non_updateable:
+            cellset_as_dict = self.drop_non_updateable_cells(cellset_as_dict, cube_name, dimensions)
```

Comment:
Thanks. I agree it should be handled one level deeper. The `write_values_through_cellset` function already takes an MDX and a list of values but the current approach to filter out the non-updateable requires a cellset dictionary structure.

I introduced a new function: `write_through_cellset` that is called from `write` and takes a cellset dictionary instead of an MDX and calls the `write_values_through_cellset`.